File: 1470.shuffle-the-array.js
/*
 * @lc app=leetcode id=1470 lang=javascript
 *
 * [1470] Shuffle the Array
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function(nums, n) {

    let i = 0;
    let j = nums.length/2;

    let shuffledArr = [];

    while(i < nums.length/2 && j < nums.length){
             
        shuffledArr.push(nums[i], nums[j]);
        i++;
        j++;
    }

    return shuffledArr;
    
};
// @lc code=end

File: 1470.shuffle-the-array.java
/*
 * @lc app=leetcode id=1470 lang=java
 *
 * [1470] Shuffle the Array
 */

// @lc code=start
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int i = 0;
        int j = nums.length/2;

        int[] ansArr = new int[nums.length];
        int k = 0;

        while(i < nums.length/2 && j < nums.length){
            ansArr[k++] = nums[i++];
            ansArr[k++] = nums[j++];
        }

        return ansArr;
    }
}
// @lc code=end

